.github/workflows/release.yml

```diff
@@ -21,12 +21,35 @@ jobs:
         with:
           fetch-depth: '0'
 
-      - name: Merge develop
+      - name: branch change and package version update
         run: |
-          git config user.name github-actions
-          git config user.email github-actions@github.com
+          git config --local user.email "github-actions@github.com"
+          git config --local user.name "GitHub Actions"
+          git checkout develop
+          npm version --no-git-tag-version --no-commit-hooks ${{ inputs.next_version }}
+
+      - name: Modify algoliaconfig file
+        run: |
+          sed -i 's/"version": ".*"/"version": "${{ inputs.next_version }}"/' cartridges/int_algolia/algoliaconfig.json
+
+      - name: Commit changes
+        run: |
+          git add .
+          git commit -m "Update version to ${{ inputs.next_version }}"
+          git push
+
+      - name: Merge develop into master
+        run: |
+          set -e
+          echo "Checking out develop branch..."
+          git checkout develop
+          git pull --ff-only
+          echo "Switching to master branch..."
           git checkout master
-          git merge origin/develop
+          git pull --ff-only
+          echo "Merging develop into master..."
+          git merge --no-ff origin/develop
```

Comment:
Why did you add `--no-ff`?
From the git doc:

> With --no-ff, create a merge commit in all cases, even when the merge could instead be resolved as a fast-forward.

This means that `master` will have additional commits and will not be in sync with develop anymore.

Reply:
Actually I've tested locally the `--no-ff` option locally and it requires a merge message, so it opens an editor.
Did you test it with a Github Action?

Reply:
Yes, I tested it here:
[this fork and logs of Github Action](https://github.com/htuzel/algoliasearch-sfcc-b2c/actions)

Reply:
I still see the merge commit: `Merge remote-tracking branch 'origin/develop'`, but I think this is because master and develop were not in sync before your test. It should works as before now :+1: